### PR TITLE
add support for eb --last-log

### PR DESCRIPTION
**Pull request by boegel**
implementation of feature request #1540

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86: please review

Basically, this allows having something like this in your bash history for easy access to the last log file:

```
vim `eb --last-log`
```

**Comment by boegel** [comment]
Hmm, actually more like this, since there's some extra output getting in the way...

```
vim `eb --last-log | grep -v '^=='`
```

File: easybuild/tools/config.py [hunk 1]
```diff
@@ -517,6 +520,32 @@ def get_log_filename(name, version, add_salt=False):
...
+    try:
+        # sorted by modification time, most recent last
+        sorted_paths = sorted(glob.glob(glob_pattern), key=lambda f: os.stat(f).st_mtime)
+    except OSError as err:
+        raise EasyBuildError("Failed to locate log files matching '%s': %s", glob_pattern, err)
+
+    # log of current session is typically listed last, should be taken into account
+    res = sorted_paths[-1]
+    if os.path.samefile(res, curlog):
+        res = sorted_paths[-2]
```
Comment (ocaisa) [inline easybuild/tools/config.py]: You could also check if the current user is the owner of the file...there could be a corner case where two EB-ers are using the same system at the same time.

Comment (boegel) [inline easybuild/tools/config.py]: Hmm, good idea, thanks.

Comment (wpoely86) [inline easybuild/tools/config.py]: :+1:

File: easybuild/tools/config.py [hunk 2]
```diff
@@ -517,6 +520,32 @@ def get_log_filename(name, version, add_salt=False):
     return filepath
 
 
+def find_last_log(curlog):
+    """Find location to last log file that is still available."""
```
Comment (wpoely86) [inline easybuild/tools/config.py]: doc args

**Comment by boegel** [comment]
@wpoely86, @ocaisa: remarks fixed, and I also got rid of the `== `-prefixed lines in the output of `--last-log`, please rereview?

You may still need to use ```vim `eb --last-log | grep 'log$'` ``` though, in case running `eb` spits out warnings to stdout through Python modules being loaded (depends on your setup)

**Comment by ocaisa** [comment]
Logic looks good to me

File: easybuild/tools/config.py [hunk 3]
```diff
@@ -488,18 +489,20 @@ def get_build_log_path():
     return res
 
 
-def get_log_filename(name, version, add_salt=False):
+def get_log_filename(name, version, date=None, timestamp=None, add_salt=False):
```
Comment (wpoely86) [inline easybuild/tools/config.py]: doc all args. And isn't nicer to append new args?

File: easybuild/tools/filetools.py [hunk 4]
```diff
@@ -971,7 +971,7 @@ def move_logs(src_logfile, target_logfile):
                              src_logfile, target_logfile, err)
 
 
-def cleanup(logfile, tempdir, testing):
+def cleanup(logfile, tempdir, testing, silent=False):
     """Cleanup the specified log file and the tmp directory, if desired."""
```
Comment (wpoely86) [inline easybuild/tools/filetools.py]: doc args

**Comment by boegel** [comment]
Last remarks fixed, going in.

Thanks for the review @ocaisa and @wpoely86!

**Pull request merged**